AK/UFixedBigInt.h

```diff
@@ -16,481 +18,414 @@
 #include <AK/StringBuilder.h>
 
 namespace AK {
+namespace Detail {
+// As noted near the declaration of StaticStorage, bit_size is more like a hint for a storage size.
+// The effective bit size is `sizeof(StaticStorage<...>) * 8`. It is a programmer's responsibility
+// to ensure that the hinted bit_size is always greater than the actual integer size.
+// That said, do not use unaligned (bit_size % 64 != 0) `UFixedBigInt`s if you do not know what you
+// are doing.
+template<size_t bit_size, typename Storage = Detail::StaticStorage<false, bit_size>>
+class UFixedBigInt;
+
+// ===== Concepts =====
+template<typename T>
+constexpr inline size_t assumed_bit_size = 0;
+template<>
+constexpr inline size_t assumed_bit_size<IntegerWrapper> = bit_width<int>;
+template<size_t bit_size>
+constexpr inline size_t assumed_bit_size<UFixedBigInt<bit_size>> = bit_size;
+template<BuiltInUFixedInt T>
+constexpr inline size_t assumed_bit_size<T> = bit_width<T>;
+}
+
+template<typename T>
+concept ConvertibleToUFixedInt = (Detail::assumed_bit_size<T> != 0);
 
 template<typename T>
-requires(sizeof(T) >= sizeof(u64) && IsUnsigned<T>) class UFixedBigInt;
+concept UFixedInt = (ConvertibleToUFixedInt<T> && !IsSame<T, Detail::IntegerWrapper>);
+
+template<typename T>
+concept NotBuiltInUFixedInt = (UFixedInt<T> && !BuiltInUFixedInt<T>);
 
 // FIXME: This breaks formatting
 // template<typename T>
 // constexpr inline bool Detail::IsIntegral<UFixedBigInt<T>> = true;
```

Comment:
This looks deletable.

Reply:
Old fixme on my side, can likely be removed
Context:
UFixedBigInt is currently marked as Unsigned and as not Signed